### PR TITLE
[SPK-64] removed ssl tricks on spark-env

### PR DESCRIPTION
**Pull request by mpenate**
## What changes were proposed in this pull request?

Removed x509 operations during spark env to generate certificates for some datastores.

## How was this patch tested?

Added unit tests and used over megadev cluster

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -149,7 +149,11 @@
       <artifactId>javax.servlet-api</artifactId>
       <version>${javaxservlet.version}</version>
     </dependency>
-
+    <dependency>
+      <groupId>ca.juliusdavies</groupId>
```
Comment (jlopezmalla) [inline core/pom.xml]: Why we need this dependency?

File: core/src/main/scala/org/apache/spark/security/SSLConfig.scala [hunk 2]
```diff
@@ -117,6 +125,52 @@ object SSLConfig extends Logging {
...
+    val bosCA = new BufferedOutputStream(new FileOutputStream(s"/tmp/$fileName"))
+    bosCA.write(formatPem(pem).getBytes)
+    bosCA.close()
+  }
+
+  // Gets raw pem from vault (without \n and folding) and outputs a well-formatted pem
+
+  def formatPem(pemRaw: String): String = {
+
+    val (begin, end) = ("-----BEGIN CERTIFICATE-----", "-----END CERTIFICATE-----")
```
Comment (jlopezmalla) [inline core/src/main/scala/org/apache/spark/security/SSLConfig.scala]: I think we have some similar methods in SSLConfig, Can we use then instead have duplicate code?

File: core/src/test/scala/org/apache/spark/security/SSLConfigTest.scala [hunk 3]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Modified in 2017 by Stratio Big Data Inc., Sucursal en España. Modifications are ©  2017  Stratio Big Data Inc., Sucursal en España.
```
Comment (jlopezmalla) [inline core/src/test/scala/org/apache/spark/security/SSLConfigTest.scala]: Is this the new Licences entry? Have we change this entry in the rest of the code?

Comment (mpenate) [reply core/src/test/scala/org/apache/spark/security/SSLConfigTest.scala]: We'll need to change this in R3

File: core/pom.xml (hunk 1 above)
Comment (Gschiavon) [reply core/pom.xml]: Same

Comment (mpenate) [reply core/pom.xml]: This is the only library with this feature.

**Pull request merged**